internal/util/util.go

```diff
@@ -219,18 +219,68 @@ func FSize(f *os.File) int64 {
 }
 
 // IsWordChar returns whether or not the string is a 'word character'
-// Word characters are defined as numbers, letters or '_'
+// Word characters are defined as numbers, letters or delimiters
 func IsWordChar(r rune) bool {
-	return unicode.IsLetter(r) || unicode.IsNumber(r) || r == '_'
+	return IsAlphanumeric(r) || IsSubwordDelimiter(r)
 }
 
 // IsNonWordChar returns whether or not the string is not a 'word character'
-// Non word characters are defined as all characters not being numbers, letters or '_'
+// Non word characters are defined as all characters not being numbers, letters or delimiters
 // See IsWordChar()
 func IsNonWordChar(r rune) bool {
 	return !IsWordChar(r)
 }
 
+// IsUpperWordChar returns whether or not the string is a 'upper word character'
+// Upper word characters are defined as numbers, upper-case letters, or delimiters
+func IsUpperWordChar(r rune) bool {
+	return IsUpperAlphanumeric(r) || IsSubwordDelimiter(r)
+}
+
+// IsLowerWordChar returns whether or not the string is a 'lower word character'
+// Lower word characters are defined as numbers, upper-case letters, or delimiters
+func IsLowerWordChar(r rune) bool {
+	return IsLowerAlphanumeric(r) || IsSubwordDelimiter(r)
+}
+
+// IsSubwordDelimiter returns whether or not the string is a 'delimiter character'
+// Delimiter characters are defined as '_'
```

Comment:
...So now that I'm looking more closely at this: the function description, like the function name, should be clear and call it a subword delimiter, not just a delimiter, right?

Maybe let's make it as clear as possible right away, e.g.: "IsSubwordDelimiter returns whether or not the string is a 'sub-word delimiter character', i.e. is considered a part of the word and is used as a delimiter between sub-words of the word. For now the only sub-word delimiter character is '_'."

BTW this is obviously not a string, this is a character aka rune. It's just a typo inherited from the `IsWordChar()` description. So it's probably better to fix this typo now, rather than spread it to all other function descriptions?

Reply:
> ...So now that I'm looking more closely at this: the function description, like the function name, should be clear and call it a subword delimiter, not just a delimiter, right?

Right.

> Maybe let's make it as clear as possible right away, e.g.: "IsSubwordDelimiter returns whether or not the string is a 'sub-word delimiter character', i.e. is considered a part of the word and is used as a delimiter between sub-words of the word. For now the only sub-word delimiter character is '_'."

So, be it.

> BTW this is obviously not a string, this is a character aka rune. It's just a typo inherited from the IsWordChar() description. So it's probably better to fix this typo now, rather than spread it to all other function descriptions?

Good catch. Yep, lets do this.

Reply:
I updated the PR accordingly to your advice and I am looking forward to your feedback.